kustomize/data_source_kustomization_overlay.go

```diff
@@ -631,6 +700,54 @@ func getKustomization(d *schema.ResourceData) (k types.Kustomization) {
 		)
 	}
 
+	if d.Get("generator_options") != nil {
+		gos := d.Get("generator_options").([]interface{})
+
+		if len(gos) == 1 && gos[0] != nil {
+			o := gos[0].(map[string]interface{})
+			k.GeneratorOptions = getGeneratorOptions(o)
+		}
+	}
+
+	if d.Get("helm_globals") != nil {
+		hgs := d.Get("helm_globals").([]interface{})
+
+		if len(hgs) == 1 && hgs[0] != nil {
+			hg := hgs[0].(map[string]interface{})
+			k.HelmGlobals = getHelmGlobals(hg)
+		}
+	}
+
+	if d.Get("helm_charts") != nil {
+		hcs := d.Get("helm_charts").([]interface{})
+		for i := range hcs {
+			if hcs[i] == nil {
+				continue
+			}
+
+			hc := hcs[i].(map[string]interface{})
+			hca := types.HelmChart{}
+
+			hca.Name = hc["name"].(string)
+			hca.Version = hc["version"].(string)
+			hca.Repo = hc["repo"].(string)
+			hca.ReleaseName = hc["release_name"].(string)
+			hca.Namespace = hc["namespace"].(string)
+			hca.ValuesFile = hc["values_file"].(string)
+			hca.ValuesMerge = hc["values_merge"].(string)
+			hca.IncludeCRDs = hc["include_crds"].(bool)
+
+			hc_vi := make(map[string]interface{})
+			err := yaml.Unmarshal([]byte(hc["values_inline"].(string)), &hc_vi)
+			if err != nil {
+				fmt.Printf("error: %v", err)
+			}
```

Comment:
Idiomatic stylistic suggestion. Hi Bradon.
```suggestion
			if err := yaml.Unmarshal([]byte(hc["values_inline"].(string)), &hc_vi); err != nil {
				fmt.Printf("error: %v", err)
			}
```
FYI - The kustomization schema should be accessible via the [openapi.SchemaForResourceType()](https://pkg.go.dev/sigs.k8s.io/kustomize/kyaml/openapi@v0.13.0#SchemaForResourceType) function, which would make it trivial to opt-in or opt-out of certain features vs. recreating the schema from scratch.

Reply:
Taken care of the small idiomatic fix. The other is a much wider change, obviously, and would go against the style of the rest of the project.